src/main/java/com/google/maps/internal/ratelimiter/LongMath.java

```diff
@@ -43,7 +43,7 @@ private LongMath() {}
    */
   public static long saturatedAdd(long a, long b) {
     long naiveSum = a + b;
-    if ((a ^ b) < 0 | (a ^ naiveSum) >= 0) {
+    if ((a ^ b) < 0 || (a ^ naiveSum) >= 0) {
```

Comment:
I’m a bit worried about this change. I think this code was imported from Guava.

Reply:
I'll look in to it.

Reply:
I believe this is the source: https://github.com/google/guava/blob/0cd4e9faa1360da4a343f84cb275d6eda0c5e732/android/guava/src/com/google/common/math/LongMath.java#L652

With that said, my feeling is this fix is correct. 

Any thoughts on this one @markmcd?

Reply:
Since the effect is the same either way, and considering this code is imported from Guava, I would leave it as it is to keep maintenance simple.

The internet suggests that the existing implementation is faster, which, if true, might explain Guava's usage. Curious, given that the `||` syntax can be short-circuited.

What's the justification for the change? If it's just that errorprone emits a warning, can you add a `@SuppressWarnings("ShortCircuitBoolean")` with a comment that this is imported code?

Reply:
> What's the justification for the change?

Just getting warning-clean under Error Prone.

> can you add a @SuppressWarnings("ShortCircuitBoolean") with a comment that this is imported code?

Yep. Will alter the PR to do so.

Reply:
I have altered this PR to do a `@SuppressWarnings("ShortCircuitBoolean")` instead of changing the actual code.